### PR TITLE
Update the network setup to reflect topology update

### PR DESCRIPTION
**Pull request by fsat**
The topology update is as such:
- Use of 3 AZs as before.
- Subnets for public agent and private agent for each AZ.
- Security groups for public agent, private agent, and core node.
- Subnet and security group for bastion host.
- Updates to ELB security group.

### CODE REVIEW

**Comment by fsat** [comment]
[As a ConductR user, I want Lightbend Prod to be deployed on 2.0, so Lightbend can find issues before me.](https://trello.com/c/6lKCCRAF/412-as-a-conductr-user-i-want-lightbend-prod-to-be-deployed-on-2-0-so-lightbend-can-find-issues-before-me)

File: src/main/play-doc/operation/Install.md [hunk 1]
```diff
@@ -444,34 +444,140 @@ This tutorial will provide you with all the key configuration details needed to
 
 Begin by preparing the EC2 network and security environment. This tutorial uses a Virtual Private Cloud (VPC) with a Classless Inter-Domain Routing (CIDR) of `10.0.0.0/16` and example addresses will be based accordingly.
```
Comment (edwardcallahan) [inline src/main/play-doc/operation/Install.md:446]: Like the Ansible scripts, many should use the single SG 'flat' cluster to try ConductR in dev/test mode. The three SG model is less complicated and a better default. Can we do similar without making this super complicated?

File: src/main/play-doc/operation/Install.md [hunk 2]
```diff
@@ -444,34 +444,140 @@ This tutorial will provide you with all the key configuration details needed to
...
-| SSH     |TCP     | 22          | 0.0.0.0/0  |		
+| Type    | Proto   | Port        | Destination | Description                |
+| :------ | :-----  | :---------- | :---------- | :------------------------- |
+| HTTP    | TCP     | 9000        | 0.0.0.0/0   | HTTP application endpoints |
+| HTTP    | TCP     | 9009        | 0.0.0.0/0   | ELB health check           |
+| HTTP    | TCP     | 9999        | 0.0.0.0/0   | Visualizer                 |
+
+#### Bastion Host Subnet and Security Group
+
+Create a subnet in the closest AZ. The bastion host will be hosted in this subnet, and thus only allowing SSH access from the outside world.
```
Comment (edwardcallahan) [inline src/main/play-doc/operation/Install.md:470]: closest AZ doesn't make sense.

File: src/main/play-doc/operation/Install.md [hunk 3]
```diff
@@ -444,34 +444,140 @@ This tutorial will provide you with all the key configuration details needed to
...
-| :------ | :-----  | :---------- | :--------- |
-| HTTP    | TCP     | 80          | 0.0.0.0/0  |
-| HTTPS   | TCP     | 443         | 0.0.0.0/0  |
+| Type    | Proto   | Port        | Source     | Description         |
+| :------ | :-----  | :---------- | :--------- | :------------------ |
+| HTTP    | TCP     | 80          | 0.0.0.0/0  | HTTP Public Access  |
+| HTTPS   | TCP     | 443         | 0.0.0.0/0  | HTTPS Public Access |
 
-SG-Nodes Inbound Rules
+SG-ELB Outbound Rules
```
Comment (edwardcallahan) [inline src/main/play-doc/operation/Install.md:460]: Why introduce outbound rules?

File: src/main/play-doc/operation/Install.md [hunk 4]
```diff
@@ -444,34 +444,140 @@ This tutorial will provide you with all the key configuration details needed to
...
-| Custom  |TCP     | 9004-9008   | SG-Nodes   |
-| Custom  |TCP     | 10000-10999 | SG-Nodes   |
-| SSH     |TCP     | 22          | 0.0.0.0/0  |		
+| Type    | Proto   | Port        | Destination | Description                |
+| :------ | :-----  | :---------- | :---------- | :------------------------- |
+| HTTP    | TCP     | 9000        | 0.0.0.0/0   | HTTP application endpoints |
+| HTTP    | TCP     | 9009        | 0.0.0.0/0   | ELB health check           |
+| HTTP    | TCP     | 9999        | 0.0.0.0/0   | Visualizer                 |
+
+#### Bastion Host Subnet and Security Group
```
Comment (edwardcallahan) [inline src/main/play-doc/operation/Install.md:468]: Optional, no?

File: src/main/play-doc/operation/Install.md (hunk 2 above)
Comment (edwardcallahan) [reply src/main/play-doc/operation/Install.md:470]: Can we use public node subnets for the bastion w/ a bastion specific security group? IOW,  why do we want a new subnet for the single node?

File: src/main/play-doc/operation/Install.md [hunk 5]
```diff
@@ -444,34 +444,140 @@ This tutorial will provide you with all the key configuration details needed to
...
+| HTTP    | TCP     | 80          | 0.0.0.0/0        | Internet Access       |
+| HTTPS   | TCP     | 443         | 0.0.0.0/0        | Internet Access       |
+
+
+#### Public Subnets and Security Group
+
+Create a subnet in each AZ. This subnet will be a public subnet, and thus accessible from outside world via ELB. The ConductR public agents will be hosted in this subnet.
+
+Our example subnets names indicate their AZ. In this case the subnet will be called SN-PUBLIC-A with a CIDR of `10.0.1.100/124`, SN-PRIVATE-B with a CIDR of `10.0.2.100/124`, and SN-PRIVATE-C with a CIDR of `10.0.3.100/124`. Each subnet will need an internet gateway added to their route table for the destination `0.0.0.0/0`. This required so that our nodes can access the internet.
+
```
Comment (edwardcallahan) [inline src/main/play-doc/operation/Install.md:499]: If I am using public and private subnets don't I need one public and one private per AZ? Having core and private agents in a private subnet (outbound IGW access only) makes sense, but a single private SN per AZ is good for both, no? Of course if users have needs, but I'm not sure what the common user gains.

File: src/main/play-doc/operation/Install.md [hunk 6]
```diff
@@ -444,34 +444,140 @@ This tutorial will provide you with all the key configuration details needed to
...
+Create security group called SG-AGENT-PUBLIC which will provide access control to ConductR public agents.
+
+SG-AGENT-PUBLIC Inbound Rules
```
Comment (edwardcallahan) [inline src/main/play-doc/operation/Install.md:502]: port 80? Isn't that what port I get with the hostname http endpoint?

File: src/main/play-doc/operation/Install.md [hunk 7]
```diff
@@ -444,34 +444,140 @@ This tutorial will provide you with all the key configuration details needed to
...
+| SSH     | TCP     | 22          | 0.0.0.0/0        | SSH to ConductR nodes |
+| HTTP    | TCP     | 80          | 0.0.0.0/0        | Internet Access       |
+| HTTPS   | TCP     | 443         | 0.0.0.0/0        | Internet Access       |
+
+
+#### Public Subnets and Security Group
+
+Create a subnet in each AZ. This subnet will be a public subnet, and thus accessible from outside world via ELB. The ConductR public agents will be hosted in this subnet.
+
+Our example subnets names indicate their AZ. In this case the subnet will be called SN-PUBLIC-A with a CIDR of `10.0.1.100/124`, SN-PRIVATE-B with a CIDR of `10.0.2.100/124`, and SN-PRIVATE-C with a CIDR of `10.0.3.100/124`. Each subnet will need an internet gateway added to their route table for the destination `0.0.0.0/0`. This required so that our nodes can access the internet.
```
Comment (edwardcallahan) [inline src/main/play-doc/operation/Install.md:498]: ' SN-PRIVATE-C ' is incorrect here. What about creating all the subnets in one go? I don't like being bounced around w/in the console when following instructions.

File: src/main/play-doc/operation/Install.md [hunk 8]
```diff
@@ -444,34 +444,140 @@ This tutorial will provide you with all the key configuration details needed to
...
+| Custom  | TCP     | 9006        | 0.0.0.0/0        | ConductR Bundle Stream Server    |
+| Custom  | TCP     | 9007        | 0.0.0.0/0        | ConductR Status Server           |
+| Custom  | TCP     | 9008        | 0.0.0.0/0        | ConductR Service Locator         |
+
+
+#### Private Subnets
+
+Create a subnet in each AZ. This subnet will be private, and thus not directly accessible from outside world. The ConductR core and the ConductR private agents will be hosted in this subnet.
+
+Our example subnets names indicate their AZ. In this case the subnet will be called SN-PRIVATE-A with a CIDR of `10.0.1.0/49`, SN-PRIVATE-B with a CIDR of `10.0.2.0/49`, and SN-PRIVATE-C with a CIDR of `10.0.3.0/49`. Each subnet will need an internet gateway added to their route table for the destination `0.0.0.0/0`. This required so that our nodes can access the internet.
```
Comment (edwardcallahan) [inline src/main/play-doc/operation/Install.md:529]: Cool. This is what I expected.

File: src/main/play-doc/operation/Install.md [hunk 9]
```diff
@@ -444,34 +444,140 @@ This tutorial will provide you with all the key configuration details needed to
...
+
+SG-AGENT-PRIVATE Inbound Rules
+
+| Type    | Proto   | Port        | Source          | Description                                   |
+| :------ | :-----  | :---------- | :-------------- | :-------------------------------------------- |
+| SSH     | TCP     | 22          | SG-BASTION      | Remote SSH Access                             |
+| Custom  | TCP     | 2552        | SG-CORE         | Akka Remoting from Core to Agent              |
+| HTTP    | TCP     | 10000-10999 | SG-AGENT-PUBLIC | Proxy access from ConductR public agent nodes |
+
+SG-AGENT-PRIVATE Outbound Rules
```
Comment (edwardcallahan) [inline src/main/play-doc/operation/Install.md:541]: Outbound rules. Still don't think we want them. I also presume we'd need a deny to activate?

File: src/main/play-doc/operation/Install.md (hunk 8 above)
Comment (edwardcallahan) [inline src/main/play-doc/operation/Install.md:529]: A single IGW for all?

**Comment by edwardcallahan** [comment]
Thanks for updating all the port mappings!

**Review by edwardcallahan — CHANGES_REQUESTED** [review]

**Comment by huntc** [comment]
@fsat can this be merged?

**Comment by huntc** [comment]
Commit 1b8ec904ea7f5f555df412720404531fa5f3debf takes care of 2.0

**Pull request merged**